java/src/jmri/server/json/JsonNamedBeanSocketService.java

```diff
@@ -40,9 +42,22 @@ public JsonNamedBeanSocketService(JsonConnection connection, H service) {
     }
 
     @Override
-    public void onMessage(String type, JsonNode data, String method, Locale locale) throws IOException, JmriException, JsonException {
+    public void onMessage(String type, JsonNode data, String method, Locale locale)
+            throws IOException, JmriException, JsonException {
         setLocale(locale);
         String name = data.path(NAME).asText();
+        // protect against a request made with a user name instead of a system name
+        if (!method.equals(PUT) && service.getManager().validSystemNameFormat(name) != NameValidity.VALID) {
+            T bean = service.getManager().getBeanBySystemName(name);
```

Comment:
Is it possible to get anything except `null` from this statement? The `if` statement ensures that the name is invalid or valid as prefix only.

Reply:
This *should* only result in `null`, but since `validSystemNameFormat(name)` in AbstractManager *always* return `VALID` (deliberately), this is programmed against the possibility the if statement is incorrectly true.